metapool/igm.py

```diff
@@ -0,0 +1,155 @@
+import os
+
+from datetime import datetime
+
+from openpyxl import load_workbook
+from openpyxl.styles import PatternFill
+
+_IGM_ROW_OFFSET = 25
+
+
+def _property_maker(obj, name, cell, default):
+    def fget(self):
+        return getattr(self, '_' + name)
+
+    def fset(self, value):
+        if value is not None:
+            self._sheet[cell] = value
+        setattr(self, '_' + name, value)
+
+    # set the default values
+    setattr(obj, '_' + name, default)
+    if default is not None:
+        getattr(obj, '_sheet')[cell] = default
+
+    setattr(type(obj), name, property(fget, fset))
+
+
+class IGMManifest(object):
+    """Interface to write an IGM manifest file
+
+    Attributes
+    ----------
+    submission_date: default today's date
+        When the sample sheet is submitted.
+    institute: default Knight Lab
+        Institution or laboratory.
+    pi_name: default Dr. Knight
+        Name of the PI.
+    pi_email: default mackkenzie.m.bryant@gmail.com
+        Email for the PI
+    contact_name: default MacKenzie Bryant
+        Contact's name.
+    project_number:
+        Project's number.
+    task_number:
+        Tasks' number
+    platform: default NovaSeq S4:
+        Sequencing platform.
+    run_type: default PE150
+        Sequencing run type.
+    custom_primer:
+        Notes about the primers in the run.
+    number_of_samples:
+        Samples in the run.
+    number_of_lanes:
+        Samples in the lane.
+    """
+    def __init__(self):
+        self._workbook = _load_igm_template()
+        self._sheet = self._workbook['Sample Information']
+
+        _property_maker(self, 'submission_date', 'B2',
+                        datetime.strftime(datetime.today(), '%m/%d/%y'))
+        _property_maker(self, 'institute', 'B3', 'Knight Lab')
+        _property_maker(self, 'pi_name', 'B4', 'Dr. Knight')
+        _property_maker(self, 'pi_email', 'B5', 'mackenzie.m.bryant@gmail.com')
```

Comment:
Should this values be global? Fine if not, just confirming.

Reply:
Good call, I had already had a typo that @charles-cowart so I've made all the defaults globals.